### PR TITLE
Upstream Debian patches

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This takes the three patches made to VAST for Debian packaging by @satta and adapts them slightly.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

@satta it'd be lovely if you could test this locally (especially for the robin map change) and publicly comment that I am allowed to use your patches for this purpose, just to be on the safe side.

@tenzir/backend Commit-by-commit. The changes are quite simple.

### CODE REVIEW

File: libvast/CMakeLists.txt [hunk 1]
```diff
@@ -4,24 +4,29 @@
...
-    "${CMAKE_CURRENT_BINARY_DIR}/tsl-robin-mapTargets.cmake"
-    "${CMAKE_BINARY_DIR}/tsl-robin-mapTargets.cmake"
-  COMMENT "Linking tsl-robin-map targets")
-install(
-  EXPORT tsl-robin-mapTargets
-  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/vast"
-  NAMESPACE tsl::)
-string(APPEND VAST_EXTRA_TARGETS_FILES
-       "\ninclude(\"\${CMAKE_CURRENT_LIST_DIR}/tsl-robin-mapTargets.cmake\")")
+find_package(tsl-robin-map 0.6.2 QUIET)
```
Comment (tobim) [inline libvast/CMakeLists.txt:7]: Please add a `VAST_ENABLE_BUNDLED_ROBIN_MAP` option.

File: tools/lsvast/CMakeLists.txt [hunk 2]
```diff
@@ -9,3 +9,19 @@ endif ()
 add_executable(lsvast lsvast.cpp)
 target_link_libraries(lsvast PRIVATE vast::libvast vast::internal)
 install(TARGETS lsvast DESTINATION "${CMAKE_INSTALL_BINDIR}")
+
+find_program(ASCIIDOCTOR asciidoctor)
```
Comment (tobim) [inline tools/lsvast/CMakeLists.txt:13]: The input looks very simple, can't we use pandoc instead and make adjustments to the source if necessary?

**Review by tobim — CHANGES_REQUESTED** [review]

File: tools/lsvast/CMakeLists.txt (hunk 2 above)
Comment (dominiklohmann) [reply tools/lsvast/CMakeLists.txt:13]: pandoc doesn't render as nicely, I tried this first. There's a massive difference in quality. I would rather do it the other way around and switch to using asciidoctor for vast.1 as well.

File: libvast/CMakeLists.txt (hunk 1 above)
Comment (dominiklohmann) [reply libvast/CMakeLists.txt]: Done.

File: tools/lsvast/CMakeLists.txt (hunk 2 above)
Comment (tobim) [reply tools/lsvast/CMakeLists.txt:13]: Then let's not add a new documentation format just like that and rewrite this page in Markdown.
Switching to adoc should be discussed properly first.

**Comment by satta** [comment]
> @satta it'd be lovely if you could test this locally (especially for the robin map change) and publicly comment that I am allowed to use your patches for this purpose, just to be on the safe side.

Surely you are, I'd be happy to see this upstream. Tested the robinmap patch, and building fine for me (I added `-DVAST_ENABLE_BUNDLED_ROBIN_MAP=OFF` to the build parameters in the package for the sake of completeness).

File: tools/lsvast/CMakeLists.txt (hunk 2 above)
Comment (dominiklohmann) [reply tools/lsvast/CMakeLists.txt]: Please check back.

**Comment by dominiklohmann** [comment]
> I added `-DVAST_ENABLE_BUNDLED_ROBIN_MAP=OFF` to the build parameters in the package for the sake of completeness.

Good idea. It's the default, but it doesn't hurt to specify explicitly.

**Review by tobim — APPROVED** [review]
Thanks @satta for the generous contribution!

**Comment by dominiklohmann** [comment]
macOS CI seems to be hanging, merging regardless.

**Pull request merged**